online-shop/src/app/authService.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { Credentials } from './credentials';
+import { HttpClient } from '@angular/common/http';
+import { User } from './user';
+import { ActivatedRoute, Router } from '@angular/router';
+
+
+@Injectable({ providedIn: 'root' })
+export class AuthService {
+    constructor(private http: HttpClient,
+        private route: ActivatedRoute,
+        private router: Router,) { }
+    user1!: User;
+    error = '';
+    
+    login(credentials: Credentials): Observable<User>
+    {
+        this.http.post<User>('http://localhost:3000/login',credentials)
```

Comment:
base url for the backed api(http://localhost:3000) should be added in a variable in the environment folder.
this way, you can have difeerent backend API's for different environments and the code looks cleaner